### PR TITLE
feat: Handle Network Errors

### LINKED ISSUE #123: enha: Handle Network Errors
- Show some user info if something unexpected happens (Timeouts/Flight Mode).
- Make sure the sentry Flutter SDK is handling/logging those cases.

### PR DESCRIPTION
**Pull request by denrase**
# Overview

- Introduce `SentrySdkMiddleware` and capture api errors.
- Show error state if main organizations/projects call fails.

Closes #123 

# Testing

- Enable flight mode and doing a reload on the main screen.

# Assets

Bildschirmfoto 2021-02-23 um 17 51 48

### CODE REVIEW

**Comment by bruno-garcia** [comment]
@denrase @maciejwalkowiak @marandaneto 

Can we drop the `Enha` and stick with conventional git? 

https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional#type-enum


`feat`, `fix` etc?

Or maybe drop these altogether? There's no hard rule to use this outside of getsentry/sentry afaik

File: lib/main.dart [hunk 1]
```diff
@@ -10,11 +10,13 @@ import 'package:sentry_flutter/sentry_flutter.dart';
...
 import 'redux/actions.dart';
-import 'redux/middlewares.dart';
+import 'redux/middleware/local_storage_middleware.dart';
+import 'redux/middleware/sentry_api_middleware.dart';
+import 'redux/middleware/sentry_sdk_middleware.dart';
 import 'redux/reducers.dart';
 import 'redux/state/app_state.dart';
 import 'screens/main/main_screen.dart';
-import 'screens/oboarding/onboarding_screen.dart';
+import 'screens/oboarding/onboarding_screen.dart'; // TODO(denis): Fix typo when onboarding PR is merged
```
Comment (HazAT) [inline lib/main.dart:19]: Just a reminder

File: lib/screens/health/health_screen.dart [hunk 2]
```diff
@@ -52,6 +48,13 @@ class _HealthScreenState extends State<HealthScreen> {
           }
         }
       );
+    } else if (viewModel.showErrorScreen) {
+      return EmptyScreen(
+          title: 'Error',
+          text: 'Something went wrong. Please try it again.',
```
Comment (HazAT) [inline lib/screens/health/health_screen.dart:54]:
```suggestion
          text: 'Something went wrong. Please try again.',
```

**Review by HazAT — CHANGES_REQUESTED** [review]

File: lib/redux/middleware/sentry_sdk_middleware.dart [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+import '../../redux/actions.dart';
+import '../../redux/state/app_state.dart';
+
+class SentrySdkMiddleware extends MiddlewareClass<AppState> {
+
+  @override
+  dynamic call(Store<AppState> store, dynamic action, NextDispatcher next) async {
+    if (action is ApiFailureAction) {
+      Sentry.captureException(action.error);
```
Comment (bruno-garcia) [inline lib/redux/middleware/sentry_sdk_middleware.dart:12]: Is the information about what action failed all in `action.error`?
Is the stack trace of the action that failed going into the event? `captureException` takes the stack trace too.

Comment (denrase) [reply lib/redux/middleware/sentry_sdk_middleware.dart:12]: You are right, i refactored the API client and appropriate action so we have this info. 👍

File: lib/main.dart (hunk 1 above)
Comment (denrase) [reply lib/main.dart]: Fixed.

**Review by bruno-garcia — APPROVED** [review]
RSLGTM

**Comment by denrase** [comment]
@bruno-garcia Had to Google RSLGTM ^^

**Pull request merged**